### ISSUE TITLE
`mx -V clean` fails in graal/compiler

### ISSUE THREAD

**Issue by zakkak** [graalvm/mx#231]
It looks like since https://github.com/graalvm/mx/commit/ca183078ca65a7135d46b9859d1dce412aa32d8e `mx -V clean` fails in graal/compiler

**To reproduce:**
```
cd tmp
git clone --depth 1 https://github.com/graalvm/mx
git clone --depth 1 https://github.com/oracle/graal
./mx/mx -V --primary-suite-path=graal/compiler clean 
```

**Output:**
```
Loading suite mx
SourceSuite.__init__(/tmp/mx/mx.mx), got vc=<__main__.GitConfig object at 0x7f31a6bd8f50>, vc_dir=/tmp/mx
OS variant detected: none
0:00:00.051149 [suite-discovery] Starting discovery with primary dir /tmp/graal/compiler/mx.compiler
SourceSuite.__init__(/tmp/graal/compiler/mx.compiler), got vc=<__main__.GitConfig object at 0x7f31a6bd8f50>, vc_dir=/tmp/graal
set_primary_dir(SiblingSuiteModel(/tmp/graal/compiler), /tmp/graal/compiler)
self._suiteRootDir = /tmp
0:00:00.192194 [suite-discovery] Adding compiler -> truffle in worklist after discovering compiler
0:00:00.192285 [suite-discovery] Adding compiler -> java-benchmarks in worklist after discovering compiler
find_suite_dir(SiblingSuiteModel(/tmp), truffle)
SourceSuite.__init__(/tmp/graal/truffle/mx.truffle), got vc=<__main__.GitConfig object at 0x7f31a6bd8f50>, vc_dir=/tmp/graal
0:00:00.349847 [suite-discovery] Discovered truffle from compiler (/tmp/graal/truffle, newly cloned: False)
0:00:00.350007 [suite-discovery] Adding truffle -> sdk in worklist after discovering truffle
find_suite_dir(SiblingSuiteModel(/tmp), java-benchmarks)
SourceSuite.__init__(/tmp/graal/java-benchmarks/mx.java-benchmarks), got vc=<__main__.GitConfig object at 0x7f31a6bd8f50>, vc_dir=/tmp/graal
0:00:00.381037 [suite-discovery] Discovered java-benchmarks from compiler (/tmp/graal/java-benchmarks, newly cloned: False)
find_suite_dir(SiblingSuiteModel(/tmp), sdk)
SourceSuite.__init__(/tmp/graal/sdk/mx.sdk), got vc=<__main__.GitConfig object at 0x7f31a6bd8f50>, vc_dir=/tmp/graal
0:00:00.423212 [suite-discovery] Discovered sdk from truffle (/tmp/graal/sdk, newly cloned: False)
0:00:00.423426 [suite-discovery] Discovery finished
Loading suite sdk
Component Polyglot Launcher does not specify dependencies
Loading suite truffle
Loading suite java-benchmarks
Loading suite compiler
Traceback (most recent call last):
  File "/tmp/mx/mx.py", line 17348, in <module>
    main()
  File "/tmp/mx/mx.py", line 17231, in main
    primary = _discover_suites(primarySuiteMxDir, load=should_load_suites)
  File "/tmp/mx/mx.py", line 3526, in _discover_suites
    _register_visit(primary)
  File "/tmp/mx/mx.py", line 3524, in _register_visit
    s._load()
  File "/tmp/mx/mx.py", line 1619, in _load
    self._load_extensions()
  File "/tmp/mx/mx.py", line 2041, in _load_extensions
    mod = __import__(extensionsName)
  File "/tmp/graal/compiler/mx.compiler/mx_compiler.py", line 149, in <module>
    _check_jvmci_version(jdk)
  File "/tmp/graal/compiler/mx.compiler/mx_compiler.py", line 145, in _check_jvmci_version
    mx.run([jdk.javac, '-d', sdu.directory, unqualified_source_path])
  File "/tmp/graal/compiler/mx.compiler/mx_compiler.py", line 145, in _check_jvmci_version
    mx.run([jdk.javac, '-d', sdu.directory, unqualified_source_path])
  File "/tmp/mx/mx.py", line 12798, in run
    s += 'env -i ' + ' '.join([n + '=' + pipes.quote(v) for n, v in env]) + ' \\' + os.linesep
ValueError: too many values to unpack
```

**Comment by dougxc** [comment]
Here's the fix:
```
diff --git a/mx.py b/mx.py
index 9e4179a..51a14b5 100755
--- a/mx.py
+++ b/mx.py
@@ -12821,7 +12821,7 @@ def run(args, nonZeroIsFatal=True, out=None, err=None, cwd=None, timeout=None, e
                 working_directory = _original_directory
             s += '# Directory: ' + os.path.abspath(working_directory) + os.linesep
         if _opts.very_verbose:
-            s += 'env -i ' + ' '.join([n + '=' + pipes.quote(v) for n, v in env]) + ' \\' + os.linesep
+            s += 'env -i ' + ' '.join([n + '=' + pipes.quote(v) for n, v in env.items()]) + ' \\' + os.linesep
         else:
             env_diff = [(k, env[k]) for k in env if k not in _original_environ]
             if env_diff:
```

I'll integrate it to mx asap.

**Comment by dougxc** [comment]
Should be fixed by 6da7f71646a.

**Comment by zakkak** [comment]
Thanks for the quick fix @dougxc
